.github/workflows/update_changelog.yml

```
@@ -0,0 +1,136 @@
+name: Update Changelog with new commit
+
+permissions:
+  pull-requests: write
+  contents: write
+
+on:
+  pull_request:
+    # Inputs the workflow accepts.
+    types: [closed]
+
+jobs:
+  pull-commit-message:
+    if: github.event.pull_request.merged == true
+    runs-on: ubuntu-latest
+    outputs:
+      message: ${{ steps.pull.outputs.message }}
+    steps:
+      - uses: actions/checkout@v4
```

We need to use the authorised version, currently:
```suggestion
      - uses: actions/checkout@11bd71901bbe5b1630ceea73d27597364c9af683 # v4.2.2
```